Developed benchmarks and generated weights for pallet gated-marketplace

Dreyhh opened the pull request:
# Benchmarks and weights for gated marketplace

## Overview
* Adds benchmarks and weights for gated marketplace

### Tickets

List the tickets of the PR linked
- [ ] Fix #37

Dreyhh committed:
Developed benchmarks and generated weights for pallet gated-marketplace

sebastianmontero commented on pallets/gated-marketplace/src/benchmarking.rs:
```diff
@@ -0,0 +1,553 @@
+	}
+
+	#[benchmark]
+	fn remove_marketplace() {
```
Creo que en este caso si tendriamos que agregar el numero maximo de aplicaciones y custodians, ya que se itera cada una para borrarla

Dreyhh replied:
Parece que niguno de los storagemaps en los que se itera (en esa extrinsic) tiene definido un limite, los valores de custodians tienen limite pero no se itera sobre ellos. De todos modos checare los demas extrinsics, me parece que para el de remover offertas si aplica

sebastianmontero replied:
Tienes razon, creo que de todos modos seria importante ver como el numero de applicaciones afecta el weight del extrinsic, creo que deberiamos de agregar una variable al benchmark para el numero de applicaciones, despues podemos pasar un valor razonable a la funcion de WeightInfo, como ves?

Dreyhh replied:
Tiene sentido, seria definir pseudolimites, uno por cada SM en el que se itere, y pasarle valores razonables "VR" < a esos limites a la funcion de weight, la desventaje es que tendriamos que estar pendiente de los valores reales para actualizarlo una vez estos superen a los VR (o a los pseudolimites), pero por lo demas creo que es buena idea si queremos dejar esos SM sin limite.

Dreyhh committed:
Developed some helper functions to populate st
orage map values for gated marketplace benchmarks. Upated gated marketplace weights

Dreyhh commented:
@sebastianmontero hice los cambios para popular los valores de algunos de los storage maps en los que se itera, no hubo cambios significantes al volver a calcular los weigths, tal vez debido a que los limites que manejamos en el runtime no son grandes. Hay algunos storage maps en los que se itera directamente pero no tienen limite definido.

Dreyhh committed:
Removed unneeded iterations in remove_selected_marketplace function

Dreyhh committed:
Added pseudolimits for ApplicationsByAccount and Custodians to some gated-marketplace benchmarks, updated weights
